specification/search/data-plane/Azure.Search/preview/2023-10-01-Preview/searchindex.json

```diff
@@ -0,0 +1,2815 @@
+            "description": "Extracts answer candidates from the contents of the documents returned in response to a query expressed as a question in natural language."
+          }
+        ]
+      },
+      "description": "This parameter is only valid if the query type is 'semantic'. If set, the query returns answers extracted from key passages in the highest ranked documents. The number of answers returned can be configured by appending the pipe character '|' followed by the 'count-<number of answers>' option after the answers parameter value, such as 'extractive|count-3'. Default count is 1. The confidence threshold can be configured by appending the pipe character '|' followed by the 'threshold-<confidence threshold>' option after the answers parameter value, such as 'extractive|threshold-0.9'. Default threshold is 0.7."
+    },
+    "Captions": {
+      "type": "string",
+      "enum": [
+        "none",
+        "extractive"
+      ],
+      "x-ms-enum": {
+        "name": "QueryCaptionType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "none",
+            "name": "None",
+            "description": "Do not return captions for the query."
+          },
+          {
+            "value": "extractive",
+            "name": "Extractive",
+            "description": "Extracts captions from the matching documents that contain passages relevant to the search query."
+          }
+        ]
+      },
+      "description": "This parameter is only valid if the query type is 'semantic'. If set, the query returns captions extracted from key passages in the highest ranked documents. When Captions is set to 'extractive', highlighting is enabled by default, and can be configured by appending the pipe character '|' followed by the 'highlight-<true/false>' option, such as 'extractive|highlight-true'. Defaults to 'None'."
+    },
+    "Vector": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The name of the kind of vector query being performed.",
+          "x-nullable": false
+        },
+        "k": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of nearest neighbors to return as top hits."
+        },
+        "fields": {
+          "type": "string",
+          "description": "Vector Fields of type Collection(Edm.Single) to be included in the vector searched."
+        },
+        "exhaustive": {
+          "type": "boolean",
+          "description": "If set to true, will compute the exact nearest neighbors by searching through all the training points returning the true k nearest neighbors."
+        }
+      },
+      "required": [
+        "kind"
+      ],
+      "description": "The query parameters for vector and hybrid search queries."
+    },
+    "RawVector": {
+      "type": "object",
+      "x-ms-discriminator-value": "vector",
```

Comment:
Does this discriminator value change behaviors on the service side when performing a VectorSearch? Or is this a side-effect of making `Vector` into a polymorphic type to share the base configuration?

Reply:
Can you expand on the question? The behavior change is to dictate which "value" parameter to expect (if kind=vector then we expected the "vector" parameter to be set with a float array, while if kind=text then we expect the "text" parameter to be set with a string). In theory we could have handled this without the kind parameter since we can infer what that of query it is simply by looking at what property is set (vector vs text), however, our understanding is that we needed to make the polymorphism explicit.

Reply:
In term of actual back-end behavior change, we treat kind=text very differently from kind=vectors (text vector queries need to be vectorized while raw vectors queries do not)

Reply:
Expanding on the original question, I was asking this as in `2023-07-01-Preview` the `Vector` type would generate as the following in Java:

```java
public class Vector {
    List<Float> value;
    Integer k;
    String fields;
}
```

With the changes here to support polymorphism this would break backwards compatibility for the class as the overall structure becomes the following (ignoring new parameters for brevity):

```java
public class Vector {
    Integer k;
    String fields;
}

public class RawVector extends Vector {
    List<Float> vector;
}

public class TextVector extends Vector {
    public String text;
}
```

Reply:
Yes that is right, we changed the shape to accommodate for our new polymorphic approach (which we didn't originally plan for). The models between 07-01 and 10-01 are indeed slightly different